webhooks/src/main/java/org/openmbee/sdvc/webhooks/controllers/WebhooksController.java

```diff
@@ -0,0 +1,200 @@
+                        Webhook newWebhook = new Webhook();
+                        newWebhook.setProject(proj);
+                        newWebhook.setUri(json.getUri());
+                        webhookRepository.save(newWebhook);
+                        response.getWebhooks().add(json);
+                    }, () -> {
+                        Map<String, Object> rejection = new HashMap<>();
+                        rejection.put("message", "No project found");
+                        rejection.put("code", 404);
+                        rejection.put("request", json);
+                        rejected.add(rejection);
+                    });
+
+                } catch (BadRequestException ex) {
+                    Map<String, Object> rejection = new HashMap<>();
+                    rejection.put("message", "Org to put project under is not found");
+                    rejection.put("code", 400);
+                    rejection.put("request", json);
+                    rejected.add(rejection);
+                    continue;
+                }
+            } else {
+                Map<String, Object> rejection = new HashMap<>();
+                rejection.put("message", "No permission to change project");
+                rejection.put("code", 403);
+                rejection.put("request", json);
+                rejected.add(rejection);
+                continue;
+            }
+        }
+        return ResponseEntity.ok(response);
+    }
+
+    @DeleteMapping(value = {"", "/{projectId}/webhooks"})
+    @Transactional
+    @PreAuthorize("isAuthenticated()")
+    public ResponseEntity<? extends BaseResponse> handleDelete( @PathVariable String projectId, @RequestBody WebhookRequest webhookRequest) {
+
+        // TODO: Determine which webhook to delete somehow. Maybe require webhook id.
+
+        Set<String> uris = new HashSet<>();
+        for (WebhookJson webhookJson : webhookRequest.getWebhooks()) {
+            uris.add(webhookJson.getUri());
+        }
+
+        WebhookResponse response = new WebhookResponse();
+        List<Webhook> webhooks = webhookRepository.findAllByProject_ProjectId(projectId);
+        if (webhooks.isEmpty()) {
+            throw new NotFoundException(response.addMessage("No webhooks found for project"));
+        }
+        for (Webhook webhook : webhooks) {
+            if (uris.contains(webhook.getUri())) {
+                webhookRepository.delete(webhook);
+                response.addMessage(String.format("Webhook for project %s to %s deleted", projectId, webhook.getUri()));
+            }
+        }
+
+        return ResponseEntity.ok(response);
+    }
+
+    public Map<String, Object> convertToMap(Object obj) {
+        return om.convertValue(obj, new TypeReference<Map<String, Object>>() {});
+    }
+
+    private boolean webhookExists(WebhookJson json, String projectId) {
+        List<Webhook> projectWebhooks = webhookRepository.findAllByProject_ProjectId(projectId);
```

Comment:
why not just make a query that checks for it directly? doesn't have to go through loop each time

Reply:
It's because the only other identifier we have is the URI, and I'm not sure that querying for that is the best way to go.